tests/Cms/Pages/PageTest.php

```diff
@@ -1111,4 +1111,69 @@ public function testPageModel()
 
         Page::$models = [];
     }
+
+    public function testPageController()
+    {
+        $app = new App([
+            'roots' => [
+                'index' => __DIR__ . '/fixtures/PageTest'
+            ],
+            'site' => [
+                'children' => [
+                    [
+                        'slug' => 'test',
+                        'content' => [
+                            'title' => 'Test Title',
+                        ]
+                    ]
+                ]
+            ],
+            'controllers' => [
+                'default' => function ($page) {
+                    $page = $page->changeTitle('New Title');
+
+                    return ['page' => $page];
+                }
+            ]
+        ]);
+
+        $app->impersonate('kirby');
+        $page = $app->page('test');
+        $data = $page->controller();
+
+        $this->assertCount(4, $data);
+        $this->assertInstanceOf('Kirby\Cms\Page', $data['page']);
+        $this->assertSame('New Title', $data['page']->title()->value());
```

Comment:
The test should also test the default objects:

```suggestion
        $this->assertSame($app, $data['kirby']);
        $this->assertSame($app->site(), $data['site']);
        $this->assertSame($app->site()->children(), $data['pages']);
        $this->assertInstanceOf('Kirby\Cms\Page', $data['page']);
        $this->assertSame('New Title', $data['page']->title()->value());
```